### PR TITLE
Introduce markers for static and runtime init recorder methods

### PR DESCRIPTION
**Pull request by geoand**
These are meant to replace the comments found in various recorder
methods

### CODE REVIEW

File: extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/OpenTelemetryRecorder.java [hunk 1]
```diff
@@ -30,23 +32,23 @@ public class OpenTelemetryRecorder {
 
     public static final String OPEN_TELEMETRY_DRIVER = "io.opentelemetry.instrumentation.jdbc.OpenTelemetryDriver";
 
-    /* STATIC INIT */
+    @StaticInit
```
Comment (gsmet) [inline extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/OpenTelemetryRecorder.java:35]: I'm all for these marker annotations as I agree things have been confusing. But shouldn't have a check then?

Comment (geoand) [reply extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/OpenTelemetryRecorder.java:35]: > But shouldn't have a check then?

I am not sure I follow this one. Care to elaborate please?

Comment (gsmet) [reply extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/OpenTelemetryRecorder.java:35]: If we add a marker, I would expect us to check that this recorder method is called at the proper phase and not at runtime init (for this particular example).

**Review by radcortez — APPROVED** [review]

**Review by brunobat — APPROVED** [review]

**Comment by geoand** [comment]
Merging and I'll follow up with an annotation processor based verificatron later

**Pull request merged**